src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,52 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int length = 0;
+        int[] diagonal = new int[matrix.length];
+
+        while (length < matrix.length) {
+            diagonal[length] = matrix[length][length];
+            length++;
+        }
+
+        return matrix == null ? null : diagonal;
```

Comment:
this check doesn't make sense, because if you had matrix == null, NullPointerException would have been thrown on  matrix.length